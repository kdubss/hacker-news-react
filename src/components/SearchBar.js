import React from 'react';

const SearchBar = ({ value, onChange, children }) =>
  <form>
    { children } <input
      type="text"
      value={ value }
      onChange={ onChange }
    />
  </form>

export default SearchBar;