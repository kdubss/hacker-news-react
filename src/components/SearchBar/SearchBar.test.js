import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from './SearchBar';
import renderer from 'react-test-renderer';

describe('SearchBar component has a valid snapshot.', () => {

    it('SearchBar component renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<SearchBar />, div);
        ReactDOM.unmountComponentAtNode(div);
    });


    test('SearchBar has a valid snapshot', () => {
        const component = renderer.create(
            <SearchBar></SearchBar>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
