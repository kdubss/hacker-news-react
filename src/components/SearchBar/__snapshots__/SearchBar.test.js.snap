// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component has a valid snapshot. SearchBar has a valid snapshot 1`] = `
<form>
  <input
    type="text"
  />
  <button
    type="submit"
  />
</form>
`;
